airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -300,6 +301,16 @@ def _send(
 
         try:
             response = self._session.send(request, **request_kwargs)
+        except requests.ConnectionError as e:
```

Comment:
Should we only catch `InvalidURL` exceptions here? It looks like previously, these failures were raised and custom-handled as `InvalidURL` exceptions.